src/TDebug.cpp

```diff
@@ -122,9 +156,117 @@ TDebug& TDebug::operator<<(const QList<QString>& list)
+    }
+    static const QString validIdentifiers{"123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"};
+    static const int maxIdCount = validIdentifiers.size();
+    int tryCount = -1;
+    do {
+        if (!TDebug::smUsedIdentifiers.contains(validIdentifiers.at(++tryCount))) {
+            break;
+        }
+    } while (tryCount < maxIdCount);
+
+    QString hostName = pHost->getName();
+    if (TDebug::smUsedIdentifiers.contains(validIdentifiers.at(tryCount))) {
+        // Run out of identifers - fall back to '?'
+        QPair<QString, QChar> newIdentifier = qMakePair(hostName, QLatin1Char('?'));
+        TDebug::smIdentifierMap.insert(pHost, newIdentifier);
+    } else {
+        QPair<QString, QChar> newIdentifier = qMakePair(hostName, validIdentifiers.at(tryCount));
+        TDebug::smUsedIdentifiers.insert(validIdentifiers.at(tryCount));
+        TDebug::smIdentifierMap.insert(pHost, newIdentifier);
+    }
+    TDebug localMessage(Qt::blue, Qt::white);
+    localMessage << QStringLiteral("Profile: '%1' started.\n").arg(hostName) >> nullptr;
+    TDebug tableMessage(Qt::white, Qt::black);
+    tableMessage << TDebug::displayNewTable() >> nullptr;
+}
+
+/* static */ void TDebug::removeHost(Host* pHost)
+{
+    TDebug::smIdentifierMap.remove(pHost);
+    TDebug localMessage(Qt::darkGray, Qt::white);
+    localMessage << QStringLiteral("Profile: '%1' ended.\n").arg(pHost->getName()) >> nullptr;
+    TDebug tableMessage(Qt::white, Qt::black);
+    tableMessage << TDebug::displayNewTable() >> nullptr;
+}
+
+/* static */ QString TDebug::displayNewTable()
+{
+    // We do not translate this currently as the Central Debug Console does not
+    // get translated content - yet?
+    QStringList messageLines;
+    QMapIterator<const Host*, QPair<QString, QChar>> itIdentifier(TDebug::smIdentifierMap);
+    while (itIdentifier.hasNext()) {
+        itIdentifier.next();
+        if (itIdentifier.key()) {
+            messageLines.append(QStringLiteral(" [%1] = \"%2\"").arg(itIdentifier.value().second, itIdentifier.value().first));
+        }
+    }
+    if (messageLines.count() > 1) {
+        std::sort(messageLines.begin(), messageLines.end());
+    }
+    messageLines.prepend(QStringLiteral(" [*] = System message, not belonging to a specific profile"));
+    if (TDebug::smIdentifierMap.count() > 1) {
+        messageLines.prepend(QStringLiteral("There are %1 profiles active now, so each message from a profile will be "
+                                            "prefixed with a single character inside a pair of '['...']'s as follows:").arg(TDebug::smIdentifierMap.count()));
+    } else {
+        messageLines.prepend(QStringLiteral("There is only a single profile active right now, so only messages not from that "
+                                            "profile will be prefixed though the indicated character will be retained and "
+                                            "used should another profile be opened:"));
```

Comment:
While smIdentifierMap.count() <= 1, do not print these messages and profile names at all, yet.

Reply:
Not sure I get you? :confused: 

As a log/recording (possibly by review by someone else) of a run of Mudlet the information is needed IMO even when only **one** profile is active.

Reply:
I agree, for just one not necessary - the problem you mention isn't something we've faced from experience. Let's solve it when we do

Reply:
My comment from above was relevant here for this, so copy it here:

> Why do the Alias messages in your first screenshot wear no prefix? Your message seems to indicate [1] instead.
DO we want to differentiate system messages from profile messages also if only one profile is active as well - or not?
If not, drop the message, as it mostly creates confusion and not clarifies anything before other profiles are added.

Graphical overview:
![image](https://user-images.githubusercontent.com/117238/107806583-56305800-6d67-11eb-8705-3559aed133e0.png)

The lines "profile started" and "profile ended" can remain, but the others should vanish.

Reply:
However - in the regions that you have said that the marking are not required - "system" messages (or if a developer  uses the old form and ends a message with `>> 0;` AND the compile swallows that - I am not sure if they do or not - perhaps I should check!) will still be marked with a `[*]` whereas the profile ones will not be marked - but showing the mark *that would be used* was intended to help keep track when things are changed - perhaps the revised messages in the next commit will put it better...

Reply:
That is why I asked (and copy one more time):

> DO we want to differentiate system messages from profile messages also if only one profile is active as well - or not?

We did not settle that, yet.
Only after we decide that, can we do away with the [*] or keep it like during multiplay.

> help keep track when things are changed

Hence the "profile started/ended" remain.

Reply:
There is code in place that allows for an active Host to be renamed (not sure if it is utilised BTW) - that also will tickle this code but it is handled by this PR.

Obviously I am biased towards what I have done but to my mind it does not make sense to only mark non-profile specific messages when more than one profile is running - as they are still not specific to a profile when only one is active. Whilst currently that is only applicable to the messages notifying of changes there is nothing, even in the original code, to say that `TDebug` messages are only for the profile. OTOH we still have the limitation that one cannot ***start*** to show the CDC without having a profile at least loaded - which is another issue I want to address (eventually) - as it  makes it harder to debug profile loading problems than it needs to be.

Reply:
> it does not make sense to only mark non-profile specific messages when more than one profile is running

Just they were not differentiated in Mudlet until now. 
Only came about because you want to indicate profile number (good) when multiple are present and they are in none. 
When there is but one profile loaded, they can just as well remain undifferentiated as they were until now.

Reply:
> Just they were not differentiated in Mudlet prior.

Yeah, because there wasn't any like that before - but there are now and there could be more in the future...

Reply:
Let's cross the bridge when we get there. 
Until then, remove these 4 lines from debug until players actually start to multiplay.
➡️ YAGNI
https://medium.com/better-programming/yagni-you-aint-gonna-need-it-f9a178cd8e1
![](https://imgs.xkcd.com/comics/the_general_problem.png)